doc/sphinx/language/extensions/arguments-command.rst

```diff
@@ -50,51 +50,57 @@ Setting properties of a function's arguments
    it's not in a section.  Applying :attr:`global` within a section extends the
    effect outside the current sections and current module in which the command appears.
 
-      `/`
-         the function will be unfolded only if it's applied to at least the
-         arguments appearing before the `/`.  See :ref:`Args_effect_on_unfolding`.
+      :n:`@name {* % @scope }`
+         a *formal parameter* of the function :n:`@reference` (i.e.
+         the parameter name used in the function definition).  Unless `rename` is specified,
+         the list of :n:`@name`\s must be a prefix of the formal parameters, including all implicit
+         arguments.  `_` can be used to skip over a formal parameter.
+         This construct declares :n:`@name` as
+         non-implicit if `clear implicits` is specified or any
+         other :n:`@name` in the :cmd:`Arguments` command is declared implicit.
+         :token:`scope` can be either scope names or their delimiting
+         keys. See :ref:`binding_to_scope`.
 
-         .. exn:: The / modifier may only occur once.
+         .. exn:: To rename arguments the 'rename' flag must be specified.
             :undocumented:
 
-      `&`
-         tells the type checking algorithm to first type check the arguments
-         before the `&` and then to propagate information from that typing context
-         to type check the remaining arguments. See :ref:`bidirectionality_hints`.
+         .. exn:: Flag 'rename' expected to rename @name into @name.
+            :undocumented:
 
-         .. exn:: The & modifier may only occur once.
+         .. exn:: Arguments of section variables such as @name may not be renamed.
             :undocumented:
 
-      :n:`( ... ) {? % @scope }`
-         :n:`(@name__1 @name__2 ...)%@scope` is shorthand for :n:`@name__1%@scope @name__2%@scope ...`
+      :n:`( {+ @argument_spec } ) {* % @scope }`
+         :n:`(@name__1 @name__2 ...)%@scope__1%@scope__2...` is shorthand for
+         :n:`@name__1%@scope__1%@scope__2... @name__2%@scope__1%@scope__2... ...`
```

Comment:
```suggestion
         :n:`(@name__1 @name__2 ...){* % @scope_key }` is shorthand for
         :n:`@name__1{* % @scope_key } @name__2%{* % @scope_key } ...`
```

A bit more readable.  Can you also do this for the other 2?  And add a new item to the list `:n:`{* %scope_key }` just before this item explaining how multiple scopes are interpreted.  Similar to the wording "When multiple scopes are attached to a type, notations are interpreted in the first scope containing them, from the top of the stack." used below.  Maybe just that sentence.

Reply:
Thanks, that's indeed much better.

> And add a new item [...]

Indeed, done

I also took the opportunity to fix the scope/scope_key glitch

Reply:
> I also took the opportunity to fix the scope/scope_key glitch

I think that should be in a separate PR that explains what the "glitch" is and allow for proper discussion.  If scope names and scope keys can be used interchangeably almost everywhere, we should update the grammar/doc for all those uses consistently.

Reply:
You are right, here it is: https://github.com/coq/coq/pull/16518